util/dvsim/Deploy.py

```diff
@@ -385,7 +385,15 @@ def kill(self):
         '''
         if self.status == "D" and self.process.poll() is None:
             self.kill_remote_job()
-            self.process.kill()
+
+            # Try to kill the running process. Send SIGTERM first, wait a bit,
+            # and then send SIGKILL if it didn't work.
+            self.process.terminate()
+            try:
+                self.process.wait(timeout=0.5)
```

Comment:
I'd go for a slightly longer timeout here (2 s maybe?), 0.5 seconds isn't much for a process to properly clean up if network resources or slow file systems are involved.

Reply:
Hmm, good point. I'll bump it up to 2s.